### ISSUE TITLE
PostgreSQL Notification - explore as alternative to Proca.Server.Notify

### ISSUE THREAD

**Issue by marcinkoziej** [fixthestatusquo/proca-server#111]
- This could also make it work across many servers.
- Is it worh it? we habe BEAM/OTP
- BEAM messages have term format, PSQL needs some serializing [?]
- Distributed caching - notify mechanism lets us update caches on all nodes
-

**Comment by hiemanshu** [comment]
Using postgresql notify:

With psql and Phoenix's PubSub Module we can easily use psql's `NOTIFY` function to send messages on updates. This however has some limitations, the json serialized data has a limit of 8000 bits and will silently fail for anything larger. Below is an example for how to set it up:
https://gist.github.com/carterbryden/e7b19d1ba1ea92e241fed259fea23491

Using realtime --
https://github.com/supabase/realtime is a postgresql monitoring service that can be run independently and send events over websockets.

Both of these options will support all of the above. Handling events, optimizing listeners, distributed caching and more.

**Comment by hiemanshu** [comment]
Nebulex supports all of the above as well. It's possible to write a cache evict, and a cache regenerate function on changes to the DB. We can setup evict / regenerate on a per model basis when needed.

**Comment by marcinkoziej** [comment]
The 8kb limit might be an issue if we wanted to send both OLD and NEW values in notification.
The json "data bags" on fields and targets are limited to 5k and campaign and action_page config are not limited (which i must fix).

**Comment by hiemanshu** [comment]
Yeah. So if we want to get more than 8K data, we can either send old data, and new ID, or use something else. Realtime is built in Elixir, and looks to be decently supported, so is a good option, but also yet another microservice.
